### ISSUE TITLE
stubfile parser mishandles varargs?

### ISSUE THREAD

**Issue by dbrosoft** [typetools/checker-framework#798]
Save `va.astub` from this [gist](https://gist.github.com/dbrosoft/564bb782b1e2c2b8e764fdaa4fc68941) and run

``` sh
java -cp $CHECKERFRAMEWORK/checker/dist/checker.jar org.checkerframework.framework.stub.ToIndexFileConverter va.astub
```

to produce `va.jaif`.  The method signature should be `f([Ljava/lang/Object;)V`, but the JAIF omits the bracket, giving the signature `f(Ljava/lang/Object;)V`.

As far as I can tell, the parser simply drops varargs annotations, providing only a boolean `isVarArgs()` method on `Parameter`.  Both the places that method is used in the Checker Framework look like hacks.

**Comment by smillst** [comment]
I added a test case framework/tests/polyall/Issue798.java that does not use `org.checkerframework.framework.stub.ToIndexFileConverter`.  The stub parser does not read annotations on varargs. (`Object @HERE ...`)  This seems to be a limitation of the version of JavaParser we are using.  (See Issue #571)

**Comment by smillst** [comment]
Fixed by 8997a34.
